### PR TITLE
Update ModelClassDiagram with new fields and event model

### PR DESCRIPTION
**Pull request by Jasperabez**
- Change person to vendor and adjust fields
- Add event model
- didn't add association as not finalized
ModelClassDiagram

### CODE REVIEW

**Comment by C5hives** [comment]
Do you want to wait for Kevin's PR to be merged before finalising this PR?

File: docs/diagrams/ModelClassDiagram.puml [hunk 1]
```diff
@@ -12,14 +12,17 @@ Class AddressBook
...
+Class UniqueVendorList
+Class Vendor
+Class Description
 Class Name
 Class Phone
 Class Tag
 
+Class UniqueEventList
+Class Event
+Class Date
```
Comment (C5hives) [inline docs/diagrams/ModelClassDiagram.puml:24]: Will need to add an additional `UniqueId` class. With the implementation of associations, `Event` and `Vendor` objects will all have their own `UniqueId`.

Comment (Jasperabez) [reply docs/diagrams/ModelClassDiagram.puml:24]: aightz

**Comment by Jasperabez** [comment]
> Do you want to wait for Kevin's PR to be merged before finalising this PR?

I think I would do that bah

**Comment by Jasperabez** [comment]
Ready to merge:
ModelClassDiagram

**Comment by kevin-pek** [comment]
Should we add in the `Association` class here as well?

**Comment by C5hives** [comment]
> Shall I add in the `Association` class here as well? Since the PR has been merged in.

Yup! Will need `Assocation` and its related classes to be added in before this PR is considered complete.
@kevin-pek @Jasperabez

**Comment by Jasperabez** [comment]
> > Shall I add in the `Association` class here as well? Since the PR has been merged in.
> 
> Yup! Will need `Assocation` and its related classes to be added in before this PR is considered complete.
> @kevin-pek @Jasperabez 

Is association class referenced anywhere in model or model manager tho

**Comment by kevin-pek** [comment]
> > > Shall I add in the `Association` class here as well? Since the PR has been merged in.
> > 
> > 
> > Yup! Will need `Assocation` and its related classes to be added in before this PR is considered complete.
> > @kevin-pek @Jasperabez
> 
> Is association class referenced anywhere in model or model manager tho

It is included within the EventTory class though, through the `UniqueAssociationList`, so I believe it should be included. However, I think we should split up the `Vendor`, `Event`, `Association` and the classes representing the fields into a separate diagram, for better readability.

**Comment by kevin-pek** [comment]
Updated the diagram to include the new classes.

image

**Pull request merged**